qiskit/circuit/quantumcircuit.py

```diff
@@ -2208,33 +2215,61 @@ def cu(self, theta, phi, lam, gamma,   # pylint: disable=invalid-name
+                      DeprecationWarning, stacklevel=2)
         return self.append(U1Gate(theta), [qubit], [])
 
     def cu1(self, theta, control_qubit, target_qubit, label=None, ctrl_state=None):
         """Apply :class:`~qiskit.circuit.library.CU1Gate`."""
         from .library.standard_gates.u1 import CU1Gate
+        warnings.warn('The QuantumCircuit.cu1 method is deprecated as of 0.16.0. It will be '
+                      'removed no earlier than 3 months after the release date. You should use the '
+                      'QuantumCircuit.cp method instead, which acts identically.',
+                      DeprecationWarning, stacklevel=2)
         return self.append(CU1Gate(theta, label=label, ctrl_state=ctrl_state),
                            [control_qubit, target_qubit], [])
 
     def mcu1(self, lam, control_qubits, target_qubit):
         """Apply :class:`~qiskit.circuit.library.MCU1Gate`."""
         from .library.standard_gates.u1 import MCU1Gate
+        warnings.warn('The QuantumCircuit.mcu1 method is deprecated as of 0.16.0. It will be '
+                      'removed no earlier than 3 months after the release date. You should use the '
+                      'QuantumCircuit.mcp method instead, which acts identically.',
+                      DeprecationWarning, stacklevel=2)
         num_ctrl_qubits = len(control_qubits)
         return self.append(MCU1Gate(lam, num_ctrl_qubits), control_qubits[:] + [target_qubit], [])
 
     def u2(self, phi, lam, qubit):  # pylint: disable=invalid-name
         """Apply :class:`~qiskit.circuit.library.U2Gate`."""
         from .library.standard_gates.u2 import U2Gate
+        warnings.warn('The QuantumCircuit.u2 method is deprecated as of 0.16.0. It will be '
+                      'removed no earlier than 3 months after the release date. You can use the '
+                      'general 1-qubit gate QuantumCircuit.u instead: u2(φ,λ) = u(pi/2, φ, λ). '
+                      'Alternatively, you can decompose it in terms of QuantumCircuit.p and '
+                      'QuantumCircuit.sx: u2(φ,λ) = p(pi/2+φ) sx p(pi/2+λ) (1 pulse on hardware).',
+                      DeprecationWarning, stacklevel=2)
         return self.append(U2Gate(phi, lam), [qubit], [])
 
     def u3(self, theta, phi, lam, qubit):  # pylint: disable=invalid-name
         """Apply :class:`~qiskit.circuit.library.U3Gate`."""
         from .library.standard_gates.u3 import U3Gate
+        warnings.warn('The QuantumCircuit.u3 method is deprecated as of 0.16.0. It will be '
+                      'removed no earlier than 3 months after the release date. You should use '
+                      'QuantumCircuit.u instead, which acts identically. Alternatively, you can '
+                      'you can decompose u3 in terms of QuantumCircuit.p and QuantumCircuit.sx: '
+                      'u3(ϴ,φ,λ) = p(φ) sx p(ϴ) sx p(λ) (2 pulses on hardware).',
```

Comment:
```suggestion
                      'u3(ϴ,φ,λ) = p(φ) sxdg p(ϴ) sx p(λ) (2 pulses on hardware).',
```

Reply:
@kdk are you sure this is sxdg?

Reply:
Only from what's in the docs: https://qiskit.org/documentation/stubs/qiskit.circuit.library.U3Gate.html#qiskit.circuit.library.U3Gate . Armonk reports 'u3' as
```
[
    PulseQobjInstruction(name='parametric_pulse', t0=0, ch="d0", pulse_shape="drag", parameters={'amp': (0.3168957987824206-0.010290779535189967j), 'beta': -4.479487069498552, 'duration': 640, 'sigma': 160}),
    PulseQobjInstruction(name='fc', t0=0, ch="d0", phase="-(P2)"),
    PulseQobjInstruction(name='parametric_pulse', t0=640, ch="d0", pulse_shape="drag", parameters={'amp': (-0.3168957987824206+0.01029077953519007j), 'beta': -4.479487069498552, 'duration': 640, 'sigma': 160}),
    PulseQobjInstruction(name='fc', t0=640, ch="d0", phase="-(P0)"),
    PulseQobjInstruction(name='fc', t0=1280, ch="d0", phase="-(P1)")
]
```

Reply:
Yes just checked, you are right. Or if one wants to use SX only, then you can absorb the phase difference of SX and SXdg into the neighboring phase gates. It becomes:

`u3(ϴ,φ,λ) = p(φ+π) sx p(ϴ+π) sx p(λ)`

This is what the OneQubitEulerDecompose does.